src/librustc/hir/map/definitions.rs

```diff
@@ -91,6 +92,10 @@ impl DefPathTable {
 pub struct Definitions {
     table: DefPathTable,
     node_to_def_index: NodeMap<DefIndex>,
+    pub hir_to_def_index: HirIdMap<DefIndex>,
+    /// `DefIndex`es created by DefCollector::create_def; used only to complete the hir_to_def_index
+    /// mapping after the lowering
```

Comment:
Can you elaborate slightly on this comment? Why can't the mapping be done immediately? 

(Another alternative might be to create a larger comment, or extend one if one exists, explaining how the overall mapping is created in the first place.)